plugins/modules/proxysql_query_rules_fast_routing.py

```diff
@@ -0,0 +1,470 @@
+    type: str
+    default: ''
+  state:
+    description:
+      - When C(present) - adds the rule, when C(absent) - removes the rule.
+    type: str
+    choices: [ "present", "absent" ]
+    default: present
+  force_delete:
+    description:
+      - By default we avoid deleting more than one schedule in a single batch,
+        however if you need this behaviour and you're not concerned about the
+        schedules deleted, you can set I(force_delete) to C(True).
+    type: bool
+    default: False
+notes:
+- Supports C(check_mode).
+extends_documentation_fragment:
+- community.proxysql.proxysql.managing_config
+- community.proxysql.proxysql.connectivity
+
+'''
+
+EXAMPLES = '''
+---
+# This example adds a rule for fast routing
+
+- name: Add a rule
+  community.proxysql.proxysql_query_rules:
+    login_user: admin
+    login_password: admin
+    username: 'user_ro'
+    schemaname: 'default'
+    destination_hostgroup: 1
+    comment: 'fast route user_ro to default schema'
+    state: present
+    save_to_disk: yes
+    load_to_runtime: yes
+
+'''
+
+RETURN = '''
+stdout:
+    description: The mysql user modified or removed from proxysql.
+    returned: On create/update will return the newly modified rule, in all
+              other cases will return a list of rules that match the supplied
+              criteria.
+    type: dict
+    "sample": {
+        "changed": true,
+        "msg": "Added rule to mysql_query_rules_fast_routing",
+        "rules": [
+            {
+                "username": "user_ro",
+                "schemaname": "default",
+                "destination_hostgroup": 1,
+                "flagIN": "0",
+                "comment": ""
+            }
+        ],
+        "state": "present"
+    }
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.proxysql.plugins.module_utils.mysql import mysql_connect, mysql_driver, mysql_driver_fail_msg
+from ansible.module_utils.six import iteritems
+from ansible.module_utils._text import to_native
+
+# ===========================================
+# proxysql module specific support methods.
+#
+
+
+def perform_checks(module):
+    if module.params["login_port"] < 0 \
+       or module.params["login_port"] > 65535:
+        module.fail_json(
+            msg="login_port must be a valid unix port number (0-65535)"
+        )
+
+    if mysql_driver is None:
+        module.fail_json(msg=mysql_driver_fail_msg)
+
+
+def save_config_to_disk(cursor):
+    cursor.execute("SAVE MYSQL QUERY RULES TO DISK")
+    return True
+
+
+def load_config_to_runtime(cursor):
+    cursor.execute("LOAD MYSQL QUERY RULES TO RUNTIME")
+    return True
+
+
+class ProxyQueryRuleFastRouting(object):
+
+    def __init__(self, module):
+        self.state = module.params["state"]
+        self.force_delete = module.params["force_delete"]
+        self.save_to_disk = module.params["save_to_disk"]
+        self.load_to_runtime = module.params["load_to_runtime"]
```

Comment:
```suggestion
    def __init__(self, module):
        self.state = module.params["state"]
        self.force_delete = module.params["force_delete"]
        self.save_to_disk = module.params["save_to_disk"]
        self.load_to_runtime = module.params["load_to_runtime"]
```
When I did the same (passed the module object and parsed it inside), one very experienced developer said that it looks a bit like a magic outside (in this case, from `main`) and that it would be better to pass the parameters to the class constructor explicitly (yes, either `module` and the parameters like `module.params["..."]`). I don't insist but it sounds very sensible.
It also looks a bit confusing to me in `main` things like `result['state'] = proxysql_query_rule_fast_routing.state`. IMO result['state'] = module.params['state'] looks clearer.

Reply:
Never mind. There are a lot of other more important things to change. Basically it's a question of personal preferences. So I marked this point as resolved. Thanks